signac/contrib/utility.py

```diff
@@ -252,14 +250,13 @@ def _nested_dicts_to_dotted_keys(t, encode=_encode_tree, key=None):
         Tuples of dotted key and values e.g. ('a.b', 'c')
 
     """
-    if encode is not None:
-        t = encode(t)
-    if isinstance(t, Mapping):
-        if t:
-            for k in t:
+    d = _encode_tree(d)
```

Comment:
As mentioned in the docs, we're always expecting `d` to be a `dict`. Hence `_encode_tree` will simply return the non-encoded object, do we even need to use this then?

Reply:
The docs are not entirely clear here. We expect callers of this function to provide `d` as a dict. However, this function is called recursively, and it may provide non-dict inputs to its recursive calls. See:

```python
yield from _nested_dicts_to_dotted_keys(d[k], key=k_)
```

However, this points out to me that we may only need to call `_encode_tree` in the `else:` case below, to handle lists. And moreover, it could be written as an inline function instead of existing as a separate `_encode_tree`. I'll refactor this a bit.

Reply:
Updated in 39165c14.